wp-content/civi-extensions/goonjcustom/Civi/MaterialContributionService.php

```diff
@@ -9,11 +9,24 @@
  *
  */
 class MaterialContributionService extends AutoSubscriber {
-  // See: CiviCRM > Administer > Communications > Schedule Reminders.
-  const CONTRIBUTION_RECEIPT_REMINDER_ID = 6;
-
   const ACTIVITY_SOURCE_RECORD_TYPE_ID = 2;
 
+  /**
+   * Fetch the Material Contribution Receipt Reminder ID.
+   */
+  public static function getContributionReceiptReminderId() {
+    try {
+      $materialContributionReceiptReminder = \Civi\Api4\ActionSchedule::get(TRUE)
+        ->addWhere('name', '=', 'Material_Contribution_Receipt')
+        ->execute()->single();
+          
+      return (int) $materialContributionReceiptReminder['id'];
```

Comment:
Is type cast needed? What is the type of `id`?
```suggestion
      return $materialContributionReceiptReminder['id'];
```

Reply:
I think, we always receive an integer value, so there's no need for type casting here. I will go ahead and remove it and use your suggestion instead.